tsweb/tsweb.go

```diff
@@ -295,23 +309,52 @@ func (f ReturnHandlerFunc) ServeHTTPReturn(w http.ResponseWriter, r *http.Reques
 // Handled requests are logged using opts.Logf, as are any errors.
 // Errors are handled as specified by the Handler interface.
 func StdHandler(h ReturnHandler, opts HandlerOptions) http.Handler {
-	if opts.Now == nil {
-		opts.Now = time.Now
-	}
 	if opts.Logf == nil {
-		opts.Logf = logger.Discard
+		return Handler(h, opts)
 	}
-	return retHandler{h, opts}
+	return LogHandler(Handler(h, opts), opts)
 }
 
-// retHandler is an http.Handler that wraps a Handler and handles errors.
-type retHandler struct {
-	rh   ReturnHandler
+// LogHandler's handled requests are logged using opts.Logf, as are any errors.
+// Any ReturnHandler's requests' errors from within are added to the log.
```

Comment:
Where's ReturnHandler in this signature?

Reply:
The ReturnHandler would be nested in the first argument h, wrapped in an *errorHandler